src/main/java/com/o19s/es/ltr/query/DerivedExpressionQuery.java

```diff
@@ -34,17 +35,20 @@
 import org.apache.lucene.search.Weight;
 
 import java.io.IOException;
+import java.util.Map;
 import java.util.Objects;
 import java.util.Set;
 import java.util.function.Supplier;
 
 public class DerivedExpressionQuery extends Query {
     private final FeatureSet features;
     private final Expression expression;
+    private final Map<String, Double> queryParamValues;
```

Comment:
you need to add this to hashCode/equals I think

Reply:
done